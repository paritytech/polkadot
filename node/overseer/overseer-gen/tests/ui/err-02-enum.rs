#![allow(dead_code)]

use polkadot_overseer_gen::*;

#[derive(Default)]
struct AwesomeSubSys;

struct SigSigSig;

struct Event;

#[derive(Clone, Debug)]
struct MsgStrukt(u8);

#[overlord(signal=SigSigSig, event=Event, gen=AllMessages, error=OverseerError)]
enum Overseer {
	#[subsystem(MsgStrukt)]
	Sub0(AwesomeSubSys),
}

#[derive(Debug, Clone)]
struct DummySpawner;

struct DummyCtx;

fn main() {
	let overseer = Overseer::<_,_>::builder()
		.sub0(AwesomeSubSys::default())
		.i_like_pie(std::f64::consts::PI)
		.spawner(DummySpawner)
		.build(|| -> DummyCtx { DummyCtx } );
}
