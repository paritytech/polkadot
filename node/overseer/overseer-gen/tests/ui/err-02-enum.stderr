error: expected `struct`
  --> $DIR/err-02-enum.rs:16:1
   |
16 | enum Overseer {
   | ^^^^

error[E0433]: failed to resolve: use of undeclared type `Overseer`
  --> $DIR/err-02-enum.rs:27:17
   |
27 |     let overseer = Overseer::<_,_>::builder()
   |                    ^^^^^^^^ use of undeclared type `Overseer`
